manager/assets/modext/widgets/media/modx.browser.js

```diff
@@ -864,13 +864,15 @@ Ext.extend(MODx.browser.Window,Ext.Window,{
     }
 
     ,onSelect: function(data) {
-        var selNode = this.view.getSelectedNodes()[0];
-        var callback = this.config.onSelect || this.onSelectHandler;
-        var lookup = this.view.lookup;
-        var scope = this.config.scope;
+        var selNode = this.view.getSelectedNodes()[0],
+            callback = this.config.onSelect || this.onSelectHandler,
+            lookup = this.view.lookup,
+            scope = this.config.scope,
+            source = this.config.source;
         this.hide(this.config.animEl || null,function(){
             if(selNode && callback){
                 var data = lookup[selNode.id];
+                data.source = source;
```

Comment:
Aside from some minor formatting fixes, just a few substantive changes here:

1. With ES6+, avoid `var` declaration unless there's a specific reason to use it. Use `const`, or when a variable will later be assigned a new value `let`.
2. Added the default source, as the source will not show up in `this.config.source` until it is _changed_ in the tree.
3. Cast the source to Integer when a value is present.

```suggestion
        const selNode = this.view.getSelectedNodes()[0],
              callback = this.config.onSelect || this.onSelectHandler,
              lookup = this.view.lookup,
              scope = this.config.scope,
              source = this.config.source || MODx.config.default_media_source
        ;
        this.hide(this.config.animEl || null, function() {
            if (callback){
                const data = lookup[selNode.id];
                data.source = source ? parseInt(source) : undefined ;
```